### PR TITLE
[Mobile] Budget page actions undo notifications

### PR DESCRIPTION
**Pull request by joel-jeremy**
Built on top of https://github.com/actualbudget/actual/pull/2892's `useUndo` hook

Suggestions on the wording of the notifications are welcome :)

image
image

### CODE REVIEW

**Review by matt-fidd — APPROVED** [review]
LGTM

There are a good amount of user facing strings added here that'll need marking for translation. Not a blocker for this PR though unless you'd like to implement now.

**Comment by youngcw** [comment]
The styling on the undo button seems broken to me.  It lost the box.

File: packages/desktop-client/src/components/budget/report/ReportComponents.tsx [hunk 1]
```diff
@@ -247,6 +250,9 @@ export const CategoryMonth = memo(function CategoryMonth({
                   onMenuAction(month, 'copy-single-last', {
                     category: category.id,
                   });
+                  showUndoNotification({
+                    message: `Budget set to last month’s budget.`,
```
Comment (MikesGlitch) [inline packages/desktop-client/src/components/budget/report/ReportComponents.tsx:254]: Is it worth translating these now or in a future pr?

**Comment by joel-jeremy** [comment]
> The styling on the undo button seems broken to me. It lost the box.

This is present on the edge version. We should fix it on a separate PR.

**Review by youngcw — APPROVED** [review]
LGTM unless you want to add in translations now.

**Pull request merged**